File: javascript/thirteen.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
	var result=0;
	if (s.indexOf("IV")!=-1)result-=2;
	if (s.indexOf("IX")!=-1)result-=2;
	if (s.indexOf("XL")!=-1)result-=20;
	if (s.indexOf("XC")!=-1)result-=20;
	if (s.indexOf("CD")!=-1)result-=200;
	if (s.indexOf("CM")!=-1)result-=200;

	for (var i=0; i<s.length; i++){
		if (s[i]==='M') result+=1000;
		if (s[i]==='D') result+=500;
		if (s[i]==='C') result+=100;
		if (s[i]==='L') result+=50;
		if (s[i]==='X') result+=10;
		if (s[i]==='V') result+=5;
		if (s[i]==='I') result+=1;
	}

	return result;
};

File: src/main/java/lzhang/leetcode/Thirteen.java
package lzhang.leetcode;

public class Thirteen {
  public int romanToInt(String s){
    int result=0;
    for (int i=0; i<s.length(); i++){
      switch(s.charAt(i)){
      case 'M':
        result+=1000;
        break;
      case 'C':
        if (i!=s.length()-1){
          if (s.charAt(i+1)=='M'){
            result+=900;
            i++;
            continue;
          } else if (s.charAt(i+1)=='D'){
            result+=400;
            i++;
            continue;
          }
        }
        result+=100;
        break;
      case 'D':
        result+=500;
        break;
      case 'X':
        if (i!=s.length()-1){
          if (s.charAt(i+1)=='C'){
            result+=90;
            i++;
            continue;
          } else if (s.charAt(i+1)=='L'){
            result+=40;
            i++;
            continue;
          }
        }
        result+=10;
        break;
      case 'L':
        result+=50;
        break;
      case 'I':
        if (i<s.length()-1){
          if (s.charAt(i+1)=='X'){
            result+=9;
            i++;
            continue;
          } else if (s.charAt(i+1)=='V'){
            result+=4;
            i++;
            continue;
          }
        }
        result+=1;
        break;
      case 'V':
        result+=5;
        break;
      }
    }
    return result;
  }
}
